addon/webextension/background.js

```diff
@@ -33,140 +33,114 @@ class PersistentPageModificationEffect {
       }
 
-      /* Dynamic vertical centering */
-      [data-tooltip-position="right"],
-      [data-tooltip-position="left"] {
-        top: 50%;
-        transform: translateY(-50%);
+      /* Show the tooltip when hovering */
+      .donotdelete:hover .donotdelete-tooltip {
+        visibility: visible !important;
       }
 
-      [data-tooltip-position="top"] {
-        bottom: 100%;
-        margin-bottom: 0;
+      /* Vertical centering */
+      [data-tooltip-position] {
+        top: 50%;
+        /*
+        * transform: translateY(-50%);
+        * (if we weren't already transforming this element)
+        */
       }
 
       [data-tooltip-position="right"] {
         left: 100%;
         margin-left: 0;
       }
 
-      [data-tooltip-position="bottom"] {
-        top: 100%;
-        margin-top: 0;
-      }
-
       [data-tooltip-position="left"] {
         right: 100%;
         margin-right: 0;
-      }
-
-      /* Dynamic horizontal centering for the tooltip */
-      [data-tooltip-position="top"]:after,
-      [data-tooltip-position="bottom"]:after {
-        left: 50%;
-        margin-left: -6px;
-      }
-
-      /* Dynamic vertical centering for the tooltip */
-      [data-tooltip-position="right"]:after,
-      [data-tooltip-position="left"]:after {
-        top: 50%;
-        margin-top: -6px;
-      }
-
-      [data-tooltip-position="top"]:after {
-        bottom: 100%;
-        border-width: 6px 6px 0;
-        border-top-color: #000;
-      }
-
-      [data-tooltip-position="right"]:after {
-        left: 100%;
-        border-width: 6px 6px 6px 0;
-        border-right-color: #000;
-      }
-
-      [data-tooltip-position="bottom"]:after {
-        top: 100%;
-        border-width: 0 6px 6px;
-        border-bottom-color: #000;
-      }
-
-      [data-tooltip-position="left"]:after {
-        right: 100%;
-        border-width: 6px 0 6px 6px;
-        border-left-color: #000;
       }`;
     this.wordSet = new Set(wordArray);
     this.insertCSSOnAllTabs();
     this.addListeners();
+    this.portFromCS = null;
     this.APPLICABLE_PROTOCOLS = ["http:", "https:", "ftp:", "file:"];
     this.CSS = {
       code: CSS
     };
   }
 
   addListeners() {
-    browser.runtime.onMessage.addListener((msg, sender, sendResponse) => {
-      this.handleMessageFromContent(msg, sender, sendResponse)
-    });
+    browser.runtime.onConnect.addListener((port) => this.connected(port));
   }
 
   /**
     * msg:  one of
     *   getList  => returns this.wordSet (list of words)
     *   wordUsed => remove a word from this.wordSet
     */
-  handleMessageFromContent(msg, sender, sendResponse) {
-    switch (msg.type) {
-      case "getList":
-        sendResponse(this.wordSet);
-        break;
-      case "wordUsed":
-        // remove word from the list
-        this.wordSet.delete(msg.word);
-        break;
-      default:
-        throw new Error(`Message type not recognized: ${msg}`);
-    }
+  connected(p) {
```

Comment:
this feels so weird and abstract.  Some future world, let's get this pattern to be written better.